internal/integration/knot/knot_protocol.go

```diff
@@ -282,23 +294,23 @@ func initTimeout(deviceChan chan entities.Device, device entities.Device) {
 func (p *protocol) findDeviceById(id string) string {
 	for _, device := range p.devices {
 		if device.ID == id {
-			return device.Name
+			return device.DevEUI
 		}
 	}
 	return ""
 }
 
 //check if response was received by comparing previous state with the new one
 func (p *protocol) checkTimeout(device entities.Device) entities.Device {
-	curDevice := p.devices[device.ID]
-	if device.Error == "timeOut" {
-		if (device.State == entities.KnotNew && curDevice.State == entities.KnotWaitReg) ||
-			(device.State == entities.KnotRegistered && curDevice.State == entities.KnotWaitAuth) ||
-			(device.State == entities.KnotAuth && curDevice.State == entities.KnotWaitConfig) ||
-			(device.State == entities.KnotAlreadyReg && curDevice.State == entities.KnotWaitUnreg) {
+	curDevice := p.devices[device.DevEUI]
+	if device.Error == values.ErrorTimeOut {
+		if (device.State == values.KNoTNew && curDevice.State == values.KNoTWaitReg) ||
```

Comment:
Please extract this condition to a function with a name that describes what is being checked and returns a boolean.